arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1741,6 +1741,57 @@ void RestReplicationHandler::handleCommandRestoreView() {
 
   LOG_TOPIC(TRACE, Logger::REPLICATION) << "restoring view: "
     << nameSlice.copyString();
+
+  if (ServerState::instance()->isCoordinator()) {
```

Comment:
why is it necessary to add this if else case here and not in the RestViewHandler ?
This seems like a dangerous changes in behaviour, I know there are more places in the replication code that work like this.

Reply:
this logic was moved from TRI_vocbase_t::createView(...)
 if (ServerState::instance()->isCoordinator()) {
since TRI_vocbase_t is only responsible for the non-coordinator case